Std/Tactic/Omega/Frontend.lean

```diff
@@ -157,17 +157,30 @@ partial def asLinearComboImpl (e : Expr) : OmegaM (LinearCombo × OmegaM Expr ×
         (← mkAppM ``Int.neg_congr #[← prf])
         (← mkEqSymm neg_eval)
     pure (-l, prf', facts)
-  | (``HMul.hMul, #[_, _, _, _, n, e']) =>
-    match intCast? n with
-    | some n' =>
-      let (l, prf, facts) ← asLinearCombo e'
+  | (``HMul.hMul, #[_, _, _, _, x, y]) =>
+    -- Check if the first factor is an integer literal
```

Comment:
This still doesn't handle `1 * 2 * x`, right? My suggestion would be to evaluate both sides to `LinearCombo`, and then allow multiplication of a constant `LinearCombo` with any `LinearCombo` and vice versa.

Reply:
Okay, I've implemented this now.